AutoReplayUploader/AutoReplayUploaderPlugin.h

```diff
@@ -165,31 +165,42 @@ struct AuthKeyCheckUploadData : public HTTPRequestData
 	}
 
 	void OnRequestComplete(HTTPRequestCompleted_t* pCallback, bool failure);
-	
+
 
 	CCallResult< AuthKeyCheckUploadData, HTTPRequestCompleted_t > requestCompleteCallback;
 };
 
+enum UploadEndpoints {
+	BALLCHASING,
+	CALCULATED
+};
+
 class AutoReplayUploaderPlugin : public BakkesMod::Plugin::BakkesModPlugin
 {
 private:
 	std::string userAgent;
 	ISteamHTTP* steamHTTPInstance = NULL;
 	std::shared_ptr<bool> uploadToCalculated = std::make_shared<bool>(false);
 	std::shared_ptr<bool> uploadToBallchasing = std::make_shared<bool>(false);
-	
+	std::shared_ptr<int> templateSequence = std::make_shared<int>(0);
+
 	std::vector<HTTPRequestData*> fileUploadsInProgress;
 	std::vector<uint8> postData;
```

Comment:
Does this need to be a member variable? Is it to persist the data after we leave the function it is set in?

Reply:
I should make it clearer in the sdk, but bindTos only work one way (cvar value changed -> shared_ptr changes).

https://github.com/tynidev/AutoReplayUploader/blob/master/AutoReplayUploader/AutoReplayUploaderPlugin.cpp#L258
Here the templateSequence is incremented, but the actual cvar doesn't get updated. We need to actually increment the cvar itself (`cvarManager->getCvar(..)->setValue(..)`)

Reply:
That explains why my sequence variable isn't getting updated across loads.